src/matroskavideo.cpp

```diff
@@ -0,0 +1,800 @@
+                                                              Exiv2::MatroskaTags(0x11, "Xmp.video.SubTLang")};
+
+constexpr std::array<Exiv2::MatroskaTags, 3> codecInfo = {Exiv2::MatroskaTags(0x1, "Xmp.video.CodecInfo"),
+                                                          Exiv2::MatroskaTags(0x2, "Xmp.audio.CodecInfo"),
+                                                          Exiv2::MatroskaTags(0x11, "Xmp.video.SubTCodecInfo")};
+
+constexpr std::array<Exiv2::MatroskaTags, 2> streamRate = {Exiv2::MatroskaTags(0x1, "Xmp.video.FrameRate"),
+                                                           Exiv2::MatroskaTags(0x2, "Xmp.audio.DefaultDuration")};
+
+/*!
+  @brief Function used to calculate Tags, Tags may comprise of more than
+      one byte. The first byte calculates size of the Tag and the remaining
+      bytes are used to calculate the rest of the Tag.
+      Returns Tag Value.
+ */
+[[nodiscard]] uint64_t returnTagValue(const byte* buf, size_t size) {
+  assert(size > 0 && size <= 8);
+
+  uint64_t b0 = buf[0] & (0xff >> size);
+  uint64_t tag = b0 << ((size - 1) * 8);
+  for (size_t i = 1; i < size; ++i) {
+    tag |= static_cast<uint64_t>(buf[i]) << ((size - i - 1) * 8);
+  }
+
+  return tag;
+}
+
+/*!
+    @brief Function used to convert buffer data into numerical information,
+        information stored in BigEndian format
+ */
+[[nodiscard]] uint64_t returnValue(const byte* buf, size_t size) {
+  uint64_t ret = 0;
+  for (size_t i = 0; i < size; ++i) {
+    ret |= static_cast<uint64_t>(buf[i]) << ((size - i - 1) * 8);
+  }
+
+  return ret;
+}
+
+}  // namespace Internal
+}  // namespace Exiv2
+
+namespace Exiv2 {
+
+using namespace Exiv2::Internal;
+
+MatroskaVideo::MatroskaVideo(BasicIo::UniquePtr io) : Image(ImageType::mkv, mdNone, std::move(io)) {
+}  // MatroskaVideo::MatroskaVideo
+
+std::string MatroskaVideo::mimeType() const {
+  return "video/matroska";
+}
+
+void MatroskaVideo::writeMetadata() {
+}
+
+void MatroskaVideo::readMetadata() {
+  if (io_->open() != 0)
+    throw Error(ErrorCode::kerDataSourceOpenFailed, io_->path(), strError());
+
+  // Ensure that this is the correct image type
+  if (!isMkvType(*io_, false)) {
+    if (io_->error() || io_->eof())
+      throw Error(ErrorCode::kerFailedToReadImageData);
+    throw Error(ErrorCode::kerNotAnImage, "Matroska");
+  }
+
+  IoCloser closer(*io_);
+  clearMetadata();
+  continueTraversing_ = true;
+  height_ = width_ = 1;
+
+  xmpData_["Xmp.video.FileName"] = io_->path();
+  xmpData_["Xmp.video.FileSize"] = (double)io_->size() / (double)1048576;
+  xmpData_["Xmp.video.MimeType"] = mimeType();
+
+  while (continueTraversing_)
+    decodeBlock();
+
+  aspectRatio();
+}
+
+void MatroskaVideo::decodeBlock() {
+  byte buf[8];
+  io_->read(buf, 1);
+
+  if (io_->eof()) {
+    continueTraversing_ = false;
+    return;
+  }
+
+  uint32_t sz = findBlockSize(buf[0]);  // 0-8
```

Comment:
[suggestion] Rename of variable to `blockSize` or something similar to improve code readability.

Reply:
done here 2da2a0a8d360decd8afb688f7b751e6395942f66